cypress/robots/BaseRobot.ts

```diff
@@ -0,0 +1,232 @@
+  seesTextWithId(id: string, text: string) {
+    cy.get(`#${id}`).should("have.text", text);
+    return this;
+  }
+
+  doesNotseesTextWithId(id: string, text: string) {
+    cy.get(`#${id}`).should("not.have.text", text);
+    return this;
+  }
+
+  seesIdVisible(id: string) {
+    cy.get(`#${id}`).should("be.visible");
+    return this;
+  }
+
+  doesNotseesIdVisible(id: string) {
+    cy.get(`#${id}`).should("not.be.visible");
+    return this;
+  }
+
+  seesTextWithClass(domClass: string, text: string) {
+    cy.get(`.${domClass}`).should("have.text", text);
+    return this;
+  }
+
+  seesDomVisibleWithCustomMatcher(domlabel: string, matcher: string) {
+    cy.get(`[${domlabel}=${matcher}]`).should("be.visible");
+    return this;
+  }
+
+  seesDomVisible(domlabel: string) {
+    cy.get(domlabel).should("be.visible");
+    return this;
+  }
+
+  seesTextInChildDom(parentDom: string, childDom: string, text: string) {
+    cy.get(`${parentDom} ${childDom}`).should("contain", text);
+    return this;
+  }
+
+  seesTextWithClassAndIndex(domClass: string, index: number, text: string) {
+    cy.get(`.${domClass}`).eq(index).should("have.text", text);
+    return this;
+  }
+
+  hasLengthForDomWithClass(domClass: string, length: number) {
+    cy.get(`.${domClass}`).should("have.length", length);
+    return this;
+  }
+
+  hasLengthForDom(parentDomClass: string, childDom: string, length: number) {
+    cy.get(parentDomClass).find(childDom).should("have.length", length);
+    return this;
+  }
+
+  seesDomContainText(dom: string, text: string) {
+    cy.get(dom).should("contain", text);
+    return this;
+  }
+
+  doesNotseesDom(dom: string) {
+    cy.get(dom).should("not.be.visible");
+    return this;
+  }
+
+  seesTextInAgGridCell(rowId: string, colId: string, text: string) {
+    cy.get(`[row-id=${rowId}]`)
+      .find(`[col-id=${colId}]`)
+      .should("have.text", text);
+    return this;
+  }
+
+  seesAgGridColumnSelected(rowId: string, colId: string) {
+    cy.get(`[row-id=${rowId}]`)
+      .find(`[col-id=${colId}]`)
+      .should("have.class", "ag-cell-range-selected");
+    return this;
+  }
+
+  seesAgGridRowSelected(rowIndexId: string) {
+    cy.get(`[aria-rowindex=${rowIndexId}]`).should(
+      "have.class",
+      "ag-row-selected"
+    );
+    return this;
+  }
+
+  seesMinimumNumberOfElementsInDom(
+    dom: string,
+    childDomClass: string,
+    minimumLength: number
+  ) {
+    cy.get(dom)
+      .find(`.${childDomClass}`)
+      .should("have.length.greaterThan", minimumLength);
+    return this;
+  }
+
+  seesNumberOfElementsInDom(
+    dom: string,
+    childDomClass: string,
+    length: number
+  ) {
+    cy.get(dom).find(`.${childDomClass}`).should("have.length", length);
+    return this;
+  }
+
+  seesPathNameInUrl(path: string) {
+    cy.location("pathname").should("eq", path);
+    return this;
+  }
+
+  seesFullPathNameWithQueryParams(path: string) {
+    const HOST = Cypress.env("APP_URL");
```

Comment:
don't read env values in baserobot. as these methods should be reusable

Reply:
this is the library provided from zemoso, i did not wrote any function in this